File: modules/wem-api/src/main/java/com/enonic/wem/api/query/aggregation/AggregationQuery.java
package com.enonic.wem.api.query.aggregation;

public abstract class AggregationQuery
{
    private String name;

    @SuppressWarnings("unchecked")
    AggregationQuery( final Builder builder )
    {
        this.name = builder.name;
    }

    public String getName()
    {
        return name;
    }

    public static class Builder<T extends Builder>
    {
        private String name;

        public Builder( final String name )
        {
            this.name = name;
        }

        @SuppressWarnings("unchecked")
        public T name( final String name )
        {
            this.name = name;
            return (T) this;
        }


    }

}

File: modules/wem-webapp/src/main/webapp/admin/common/js/query/aggregation/AggregationQuery.ts
module api.query.aggregation {

    export class AggregationQuery {

        private name:string;

        constructor( name:string )
        {
            this.name = name;
        }

        public getName():string
        {
            return this.name;
        }

        public static newTermsAggregation( name:string ):TermsAggregationQueryBuilder
        {
            return new TermsAggregationQueryBuilder( name );
        }
    }

    export class AggregationQueryBuilder {

        name:string;

        constructor( name:string ) {
            this.name = name;
        }

        public setName( name:string ):AggregationQueryBuilder
        {
            this.name = name;
            return this;
        }
    }
}
